### PR TITLE
refactor: update some libanki methods and clean some things to prepare for the previewer port

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
Bunch of preparatory refactors for #14378 

## How Has This Been Tested?

CI + went around clicking things in the app to see if anything crashes

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt [hunk 1]
```diff
@@ -216,16 +215,16 @@ open class Card : Cloneable {
...
             oDid,
             flags,
-            data
+            ""
         )
         col.log(this)
     }
 
-    fun q(reload: Boolean = false, browser: Boolean = false): String {
+    fun question(reload: Boolean = false, browser: Boolean = false): String {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:223]: 👍 https://github.com/ankitects/anki/commit/f900f24f60e85f150418137b94dd1b8359be96c3

File: AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt [hunk 2]
```diff
@@ -420,16 +406,6 @@ open class SoundPlayer {
         }
     }
 
-    fun playOrPauseSound() {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt:423]: This was kept due to a revert - we do want the functionality, but it's trivial enough to add back in when we get around to it

**Review by david-allison — APPROVED** [review]
Let's keep this train moving

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt [hunk 3]
```diff
@@ -216,7 +215,7 @@ open class Card : Cloneable {
             oDue,
             oDid,
             flags,
-            data
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:219]: Dealt with. PTAL

[the remark this answers is not in the document]
Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:219]: If it's only used in tests, LGTM

**Comment by mikehardy** [comment]
trivial rebase to merge #14785 into two chunks here which were only perturbed here for spacing, pushed that, reviewing

**Review by mikehardy — APPROVED** [review]
Assuming CI passes - good to go - all concerns appear to be handled well

**Pull request merged**